docs/docs/markers.mdx

```diff
@@ -0,0 +1,291 @@
+
+Markers allow you to describe points in a dialogue that you are interested in identifying. In Rasa, a dialogue is represented as a sequence of events, which include bot actions that were executed (`ActionExecuted`), intents that were detected (`UserUttered`), and slots that were set (`SlotSet`). A point of interest in a dialogue can thus be expressed as a logical expression that can be evaluated against the sequence of dialogue events. When this logical expression is met, we "mark" that point in the dialogue for further analysis or inspection.
+
+There are several applications for Markers. For example, they can be used to define your bot's Key Performance Indicators (KPIs), such as dialogue completion or task success. For [Carbon Bot](https://rasa.com/blog/using-conversation-tags-to-measure-carbon-bots-success-rate/) (which helps users offset their carbon emissions from flying) dialogue completion can be defined as "all mandatory slots have been filled", while task success can be defined as "all mandatory slots have been filled and a carbon estimate has been successfully computed". Marking these events allows us to evaluate Carbon Bot's performance by quantifying how often it succeeds and how often it fails.
+
+Markers also allow you to diagnose your dialogues by surfacing important events for further inspection. For example, we might observe that Carbon Bot tends to successfully set the `travel_departure` and `travel_destination` slots, but not the `travel_flight_class`. We can define a marker to quantify how often this scenario occurs and surface relevant dialogues for review as part of [Conversation Driven Development (CDD)](https://rasa.com/blog/conversation-driven-development-a-better-approach-to-building-ai-assistants/).
+
+Markers are defined using a simple `yaml` syntax. For example, here are markers that define dialogue completion and task success for Carbon Bot:
+
+```yaml
+marker_dialogue_completion:
+    and:
+        - slot_was_set: travel_departure
+        - slot_was_set: travel_destination
+        - slot_was_set: travel_flight_class
+marker_task_success:
+    and:
+        - slot_was_set: travel_departure
+        - slot_was_set: travel_destination
+        - slot_was_set: travel_flight_class
+        - action: provide_carbon_estimate
+```
+
+And here is a marker for surfacing dialogues where all mandatory slots are set except `travel_flight_class`:
+
+```yaml
+marker_dialogue_completion:
+    and:
+        - slot_was_set: travel_departure
+        - slot_was_set: travel_destination
+        - not:
+            slot_was_set: travel_flight_class
+```
+
+## Defining Markers
+
+To describe the events that we're interested in, we provide `conditions` and `operators`.
+
+Conditions are simple expressions that describe an event, and are used like to variables in a logical expression, for example `action: utter_greet`. Operators allow you to combine conditions or sub-marker definitions which include nested operators and conditions.
```

Comment:
I think `event` is ambiguous here because it was used to refer to the technical Rasa notion of "event" (`ActionExecuted`, `UserUttered`, etc) at the top, but is referred in a "something that happened at some point in the dialogue" sense here.

Reply:
@aeshky 

> and are used like to variables

You meant "and are used like variable", right?

> for example `action: utter_greet`

I don't understand this example. Is `action` the condition and `utter_greet` the event?